### PR TITLE
[CI] Mutation testing: Automated github action to generate issue

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #2114

## What is the purpose of the change

1. Make github action that runs once per week
2. It runs make test-mutation
3. Takes mutation_test_result.txt, and creates a github issue. (with some formatting. Perhaps only parse out failures, and split them up with a number and separate code block)

## Brief Changelog
n/a

## Testing and Verifying
Untested because need `secrets.ADD_TO_PROJECT_PAT` access

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by stackman27** [comment]
The current implementation outputs mutation test FAILS in a single python output. I'm not sure how i'll break each of the fails  and split them up with a number and separate code block without using regex 

cc @alexanderbez @ValarDragon @nikever

File: scripts/mutest-format.py
Comment (nikever) [inline scripts/mutest-format.py:1]: If you need to skip `PASS` and `SKIP` lines you could simply do:

```bash
cat mutation_test_result.txt | grep -Ev "PASS | SKIP" | tee mutation_test_result.txt
```

Without the need of having this python script

**Comment by nikever** [comment]
You can add code blocks with `sed`.

Let's the file contains this:

```bash
--- ./x/tokenfactory/keeper/createdenom.go	2022-07-21 11:57:23.000000000 +0200
+++ /var/folders/p8/6wtxs79s3x75t51d07gnv9c00000gn/T/go-mutesting-3068611658/./x/tokenfactory/keeper/createdenom.go.0	2022-07-23 10:16:35.000000000 +0200
@@ -77,7 +77,7 @@
 	if err != nil {
 		return err
 	}
-	if len(creationFee) > 0 {
+	if len(creationFee) >= 0 {
 		if err := k.distrKeeper.FundCommunityPool(ctx, creationFee, accAddr); err != nil {
 			return err
 		}

FAIL "/var/folders/p8/6wtxs79s3x75t51d07gnv9c00000gn/T/go-mutesting-3068611658/./x/tokenfactory/keeper/createdenom.go.0" with checksum 8530d085e7d2c8848191f6d9471baeeb

```

With

```bash
cat | sed "s# @@# @@\n\`\`\`go\n#g " | sed "s#FAIL#\`\`\`\nFAIL#g " > test.diff
```

It will encapsulate the go code in go block:

```bash
--- ./x/tokenfactory/keeper/createdenom.go	2022-07-21 11:57:23.000000000 +0200
+++ /var/folders/p8/6wtxs79s3x75t51d07gnv9c00000gn/T/go-mutesting-3068611658/./x/tokenfactory/keeper/createdenom.go.0	2022-07-23 10:16:35.000000000 +0200
@@ -77,7 +77,7 @@
```go

 	if err != nil {
 		return err
 	}
-	if len(creationFee) > 0 {
+	if len(creationFee) >= 0 {
 		if err := k.distrKeeper.FundCommunityPool(ctx, creationFee, accAddr); err != nil {
 			return err
 		}

\``` # <--- I have added \ to escape the closing block
FAIL "/var/folders/p8/6wtxs79s3x75t51d07gnv9c00000gn/T/go-mutesting-3068611658/./x/tokenfactory/keeper/createdenom.go.0" with checksum 8530d085e7d2c8848191f6d9471baeeb
```

File: scripts/mutest-format.py
Comment (stackman27) [reply scripts/mutest-format.py:1]: I see thank you!

**Comment by stackman27** [comment]
Thank you @nikever for the feedback. I'm still getting this error even after generating the GITHUB_TOKEN. I'm guessing i'm gonna have to get access to the token in the OSMOSIS repo? [Link to issue](https://github.com/stackman27/osmosis/runs/7483794884?check_suite_focus=true)

Screen Shot 2022-07-23 at 3 12 15 PM

**Comment by alexanderbez** [comment]
@stackman27 do you have an example of what a github issue would look like when created?

File: .github/workflows/mutest-issue-generate.yml
Comment (alexanderbez) [inline .github/workflows/mutest-issue-generate.yml:7]: Why not off of `main`?

File: .github/workflows/mutest-issue-generate.yml
Comment (alexanderbez) [inline .github/workflows/mutest-issue-generate.yml:8]: Aren't we missing the weekly cron?

File: .github/workflows/mutest-issue-generate.yml
Comment (alexanderbez) [inline .github/workflows/mutest-issue-generate.yml:4]:
```suggestion
```

**Comment by stackman27** [comment]
> @stackman27 do you have an example of what a github issue would look like when created?

Currently, this line is failing `token: ${{ secrets.ADD_TO_PROJECT_PAT }}`  as i'm not sure i have access to creating github issues via actions 

Once resolved the issues should look like [this](https://github.com/osmosis-labs/osmosis/pull/2204#issuecomment-1193092452)

File: .github/workflows/mutest-issue-generate.yml
Comment (stackman27) [reply .github/workflows/mutest-issue-generate.yml:7]: just so that it runs every time i push to my branch. 

All of it is just for testing

**Comment by p0mvn** [comment]
> > @stackman27 do you have an example of what a github issue would look like when created?
> 
> Currently, this line is failing `token: ${{ secrets.ADD_TO_PROJECT_PAT }}` as i'm not sure i have access to creating github issues via actions
> 
> Once resolved the issues should look like [this](https://github.com/osmosis-labs/osmosis/pull/2204#issuecomment-1193092452)

It should be possible to create this token on your personal fork, merge this into the main of the fork, and test if this functions correctly by attempting to create an issue there

**Comment by ValarDragon** [comment]
Code LGTM, would like to see a sample issue before hitting merge tho

**Comment by ValarDragon** [comment]
I also think we're going to eventually need to do a more sophisticated python script to parse this and create the issue text body, but no need to block on this, getting the initial regular issue out is already awesome =)

**Comment by stackman27** [comment]
> > > @stackman27 do you have an example of what a github issue would look like when created?
> > 
> > 
> > Currently, this line is failing `token: ${{ secrets.ADD_TO_PROJECT_PAT }}` as i'm not sure i have access to creating github issues via actions
> > Once resolved the issues should look like [this](https://github.com/osmosis-labs/osmosis/pull/2204#issuecomment-1193092452)
> 
> It should be possible to create this token on your personal fork, merge this into the main of the fork, and test if this functions correctly by attempting to create an issue there

Screen Shot 2022-08-05 at 12 14 44 AM

hmm looks like i can't create issues in my osmosis-fork. I dont even have a tab for issues 🤔

**Comment by p0mvn** [comment]
> > > > @stackman27 do you have an example of what a github issue would look like when created?
> > > 
> > > 
> > > Currently, this line is failing `token: ${{ secrets.ADD_TO_PROJECT_PAT }}` as i'm not sure i have access to creating github issues via actions
> > > Once resolved the issues should look like [this](https://github.com/osmosis-labs/osmosis/pull/2204#issuecomment-1193092452)
> > 
> > 
> > It should be possible to create this token on your personal fork, merge this into the main of the fork, and test if this functions correctly by attempting to create an issue there
> 
> Screen Shot 2022-08-05 at 12 14 44 AM
> 
> hmm looks like i can't create issues in my osmosis-fork. I dont even have a tab for issues thinking

That's weird haha. It looks like forks have to manually enable issues in settings:
https://softwareengineering.stackexchange.com/questions/179468/forking-a-repo-on-github-but-allowing-new-issues-on-the-fork

**Comment by stackman27** [comment]
@ValarDragon @alexanderbez @p0mvn 
Here's what the issue looks like: https://github.com/stackman27/osmosis/issues/12

the code is supposed to remove "SKIP" but i'm not sure why it isn't 🤔

**Comment by stackman27** [comment]
@ValarDragon @p0mvn @alexanderbez Mutation testing issue sample issue: https://github.com/stackman27/osmosis/issues/13

**Review by czarcas7ic — APPROVED** [review]
This LGTM, are we switching this from testing to prod values before merging?

**Comment by stackman27** [comment]
> This LGTM, are we switching this from testing to prod values before merging?

Yup just pushed the production version! Good to go now 🚀

**Review by p0mvn — APPROVED** [review]
Nice work

**Comment by p0mvn** [comment]
Merging since test issue was shown here: https://github.com/stackman27/osmosis/issues/13

Thanks, @stackman27

**Pull request merged**
